flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/WriteSinkFunctionByMillis.java

```diff
@@ -1,53 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements.  See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License.  You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.apache.flink.streaming.api.functions.sink;
-
-import org.apache.flink.annotation.PublicEvolving;
-
-/**
- * Implementation of WriteSinkFunction. Writes tuples to file in every millis
- * milliseconds.
- *
- * @param <IN>
- *            Input tuple type
- */
-@PublicEvolving
```

Comment:
I don't think we outright delete these.

Reply:
I think we should, because they are bogus and no-one should actually use them.

There is already a follow-up issue: https://issues.apache.org/jira/browse/FLINK-9195 😅

Reply:
I agree that the implementation is horrible. It's inefficient, redundant and untested, but that's unfortunately not really the point here.

They _work_, someone _may_ use them, and they _are_ part of the public API. These methods have always been there, why are we _now_ breaking the API? Why not deprecate them and plaster the constructors with WARN log messages?

I don't see how this change is related to FLINK-9195, given that these classes are completely unused.

Reply:
I get your point. Would you be ok with opening a separate Jira issue, deprecating them now and removing in 1.6?

I was always under the impression that we would remove `@PublicEvolving` stuff if necessary but the description of `@PublicEvolving` seems to indicate something else. 😓